exonum-java-binding/core/rust/src/utils/errors.rs

```diff
@@ -125,3 +129,109 @@ pub fn describe_java_exception(env: &JNIEnv, exception: JObject) -> String {
     };
     unwrap_jni_verbose(env, format())
 }
+
+type ExceptionResult<T> = thread::Result<result::Result<T, JniError>>;
+
+/// Returns value or "throws" exception. `error_val` is returned, because exception will be thrown
+/// at the Java side. So this function should be used only for the `panic::catch_unwind` result.
+pub fn unwrap_exc_or<T>(env: &JNIEnv, res: ExceptionResult<T>, error_val: T) -> T {
+    match res {
+        Ok(val) => {
+            match val {
+                Ok(val) => val,
+                Err(jni_error) => {
+                    // Do nothing if there is a pending Java-exception that will be thrown
+                    // automatically by the JVM when the native method returns.
+                    if !env.exception_check().unwrap() {
+                        // Throw a Java exception manually in case of an internal error.
+                        throw(env, &jni_error.to_string())
+                    }
+                    error_val
+                }
+            }
+        }
+        Err(ref e) => {
+            throw(env, &any_to_string(e));
+            error_val
+        }
+    }
+}
+
+/// Same as `unwrap_exc_or` but returns default value.
+pub fn unwrap_exc_or_default<T: Default>(env: &JNIEnv, res: ExceptionResult<T>) -> T {
+    unwrap_exc_or(env, res, T::default())
+}
+
+/// Calls a corresponding `JNIEnv` method, so exception will be thrown when execution returns to
+/// the Java side.
+fn throw(env: &JNIEnv, description: &str) {
+    // We cannot throw exception from this function, so errors should be written in log instead.
+    let exception = match env.find_class("java/lang/RuntimeException") {
+        Ok(val) => val,
+        Err(e) => {
+            error!(
+                "Unable to find 'RuntimeException' class: {}",
+                e.description()
+            );
+            return;
+        }
+    };
+    if let Err(e) = env.throw_new(exception, description) {
+        error!(
+            "Unable to find 'RuntimeException' class: {}",
```

Comment:
While we are here, I'd fix this incorrect error message to "Failed to throw RuntimeException({}): {}" description, e.description

Reply:
To avoid confusion, we might rename the parameter to `message`/`error_message`.